pkg/core/chain/chain.go

```diff
@@ -197,11 +197,10 @@ func (c *Chain) Close() error {
 }
 
 func (c *Chain) onAcceptBlock(m bytes.Buffer) error {
-	// Ignore blocks from peers if we are only one behind - we are most
-	// likely just about to finalize consensus.
-	// TODO: we should probably just accept it if consensus was not
-	// started yet
-	if !c.counter.IsSyncing() {
+	// Ignore blocks from peers if we are only one behind, and in
+	// full-node mode - we are most likely just about to finalize
+	// consensus.
+	if !cfg.Get().General.WalletOnly && !c.counter.IsSyncing() {
```

Comment:
Would `cfg.Get().General.WalletOnly` here make an inconsistency between lightNode and fullNode?  To what (mis)understand full node would be one block behind the light node?

Reply:
You need to take into account the way the consensus works. As I've outlined before, the reason a full node does not accept outside blocks until it is two behind has to do with the introduction of the `intermediate block` and the `sleeper` and `dozer` mode. A full node will not need to accept blocks from other peers until it is two behind, since it will accept these blocks internally as received and validated through the consensus. When it does fall behind this far, it should immediately sync up as far as possible and then request additional info. We do not want this procedure to happen until it is absolutely necessary, hence why we wait to be behind two blocks, as slight delays between message propagation can already cause it to happen when considering the case where a full node accepts blocks from other peers that are only 1 ahead.

A full node running consensus is in actuality (assuming it is at the tip of the execution trace) one block ahead of it's light node peers, as it has the intermediate block.